flow/Trace.cpp

```diff
@@ -220,6 +221,46 @@ struct TraceLog {
 		}
 	};
 
+	struct IssuesList : ITraceLogIssuesReporter, ThreadSafeReferenceCounted<IssuesList> {
+		IssuesList(){};
+		void addAndExpire(std::string issue, double expirationInterval) override {
+			MutexHolder h(mutex);
+			auto now = ::now();
+			if (issues.find(issue) != issues.end()) {
+				issues[issue]++;
+			} else {
+				issues[issue] = 1;
+			}
+			queue.emplace_back(now + expirationInterval, issue);
```

Comment:
I think the expiration logic only works if expiration interval is fixed for the list, as otherwise an expired item may get placed behind an unexpired one. This could be resolved with something like a priority queue, or if it's intended that the expiration interval be fixed we could instead make it a property of the list rather than a parameter of this function.

Reply:
I changed the function to not taking the expiration interval parameter and added a TODO for later possible refactoring.